### PR TITLE
HT-2638: script to compare cluster for given HTID

### PR DESCRIPTION
**Pull request by aelkiss**
uses rspec matchers to compare:

- list of htids in cluster
- list of reported owners for htid
- holdings details for htid

### CODE REVIEW

File: bin/compare_clusters.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+require "rspec"
+require "rspec/expectations"
+require "rspec/matchers"
+
+include RSpec::Matchers
+
+Services.mongo!
+
+def overlap_line(overlap_hash)
```
Comment (jsteverman) [inline bin/compare_clusters.rb:22]: Not used?

**Review by jsteverman — COMMENTED** [review]
I expect this to take a very long time, but it shouldn't need to be run more than a handful of times.

File: bin/compare_clusters.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  ARGF.each_line do |line|
+    item_id = line.strip
+
+    #
+    cluster = Cluster.where("ht_items.item_id": item_id).first
+
+    new_htitems = cluster.ht_items.map(&:item_id).sort
+
+    new_holdings = {}
+    ClusterOverlap.new(Cluster.where("ht_items.item_id": item_id).first).each do |o|
```
Comment (billdueber) [inline bin/compare_clusters.rb:45]: I see `#each` is implemented in `ClusterOverlap`. If you `include Enumerable` in that class, you'll get all those methods for free (e.g., map, any?, etc.)

**Review by billdueber — APPROVED** [review]
Not really adding anything, just needed to walk through it to get a better sense of the codebase.

File: bin/compare_clusters.rb (hunk 2 above)
Comment (aelkiss) [reply bin/compare_clusters.rb:45]: :+1:

File: bin/compare_clusters.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+def overlap_line(overlap_hash)
+  [overlap_hash[:cluster_id],
+   overlap_hash[:volume_id],
+   overlap_hash[:member_id],
+   overlap_hash[:copy_count],
+   overlap_hash[:brt_count],
+   overlap_hash[:wd_count],
+   overlap_hash[:lm_count],
+   overlap_hash[:access_count]].join("\t")
+end
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply bin/compare_clusters.rb:31]: It also isn't used (see @jsteverman 's comment), so just going to remove it

**Comment by aelkiss** [comment]
Travis is being very slow for me today, but rubocop and rspec are both passing locally so I'm going to go ahead and merge.

**Pull request merged**